vadl/main/resources/templates/lcb/lld/ELF/Arch/ImmediateUtils.h

```diff
@@ -2,11 +2,13 @@
 #define LLVM_LIB_TARGET_[(${namespace})]_UTILS_IMMEDIATEUTILS_H
 
 #include <cstdint>
+#include "vadl-builtins.h"
+
 
 // collection of all available immediates
 
-«FOR immediate : immediates»
-    #include "«processorName»Immediates/«immediate.loweredImmediate.identifier».hpp"
-«ENDFOR»
+[# th:each="function : ${encodeFunctions}" ]
+static [(${function})]
```

Comment:
minor: `static` means that the encoding function will not be exported in the object file and thus if you include it then it will not available. Is that really what you want?